libnavigation-core/src/test/java/com/mapbox/navigation/core/reroute/MapboxRerouteControllerTest.kt

```diff
@@ -382,6 +382,24 @@ class MapboxRerouteControllerTest {
         }
     }
 
+    @Test
+    fun `deliver failure synchronously when options update fails`() =
```

Comment:
How does this test check that it's synchronous? You're using `runBlockingTest`: the test will pass even if it's not synchronous.

Reply:
I'm using a `pauseDispatcher` which will schedule but won't execute coroutines blocks it runs across, the test won't pass with the async code on the `main` branch.